### ISSUE TITLE
100% CPU usage on idle [SOLVED]

### ISSUE THREAD

**Issue by bytepoets-mba** [ejurgensen/forked-daapd#1176]
Hi!

I am using forked-daapd to play streaming radio to all of my HomePods (there are four: WZ, EZ, VR and SZ). Everything runs smoothly for a couple of hours but then for no apparent reason the CPU-Usage (when idle) goes up to 100% and stays at this level.
<img width="2306" alt="Screenshot 2021-01-22 at 09 59 44" src="https://user-images.githubusercontent.com/8235666/105469607-d13aad00-5c98-11eb-9553-c7a616846a9f.png">
<img width="1675" alt="Screenshot 2021-01-22 at 10 19 19" src="https://user-images.githubusercontent.com/8235666/105471791-6a6ac300-5c9b-11eb-9125-1f9965ab0cb4.png">

See also: top
<img width="1082" alt="Screenshot 2021-01-22 at 10 02 27" src="https://user-images.githubusercontent.com/8235666/105469871-2080dd80-5c99-11eb-8eaa-c0cbc2e1bef2.png">

I am running on a Raspi 3B, all with latest updates with latest FORKED-DAAPD - VERSION 27.2.

The issue goes away when I reboot the raspi - but only for a couple of hours. I also checked the other issues with high-cpu usage but unfortunately their problems/fixes do not seem to apply to me.

I attached the log with log-level SPAM and it seems to happen around [2021-01-22 04:25:52] where a lot of "web: LWS" entries appear: [forked-daapd.log](https://github.com/ejurgensen/forked-daapd/files/5854689/forked-daapd.log)

Btw, the project is really great(!) and thank you very much for your help in advance. 

Kind regards
Markus

**Comment by ejurgensen** [comment]
Happy to hear you like forked-daapd and thanks for a good bug report. Your issue is strange, from the log it seems there is no event triggering the CPU consumption. The "web: LWS" entries are made by libwebsockets, which is used by the web interface, so one question is if you had the web UI open when it happened? You can also try using "htop" to identify which thread is taking the CPU. I would expect it to be the "httpd" thread.

In the rare event that libwebsockets is somehow corrupted you could try reinstalling it.

**Comment by bytepoets-mba** [comment]
Thanks for the answer. I did not have the web interface open at all :-/ strange. Here is the htop:
<img width="2081" alt="Screenshot 2021-01-22 at 10 41 43" src="https://user-images.githubusercontent.com/8235666/105474721-dc90d700-5c9e-11eb-9be9-52bc3acffea5.png">

These are threads, right? I'm not by chance running 9 **instances** of forked-daapd?

**Comment by bytepoets-mba** [comment]
I found it - really sorry to have wasted your time.

A node-RED instance I am running seems to restart at about 04:30 AM every day and there is a flow with an `exec node` included that is listening to daapd to display the currently running song on my lametric using this code I found online:
`curl --include --no-buffer --header "Connection: Upgrade" --header "Upgrade: websocket" --header "Host: 192.168.1.94:3688" --header "Origin: http://192.168.1.94:3688" --header "Sec-WebSocket-Key: SGVsbG8sIHdvcmxkIQ==" --header "Sec-WebSocket-Version: 13" --header "Sec-WebSocket-Protocol: notify" http://192.168.1.94:3688/ --data "{ \"notify\": [ \"queue\" ] }"`

So - all my fault. Sorry again.
